code/game/machinery/kitchen/microwave.dm

```diff
@@ -46,7 +35,7 @@
 			for (var/reagent in recipe.reagents)
 				acceptable_reagents |= reagent
 			if (recipe.items)
-				max_n_of_items = max(max_n_of_items,recipe.items.len)
+				max_n_of_items = max(max_n_of_items,recipe.count_n_items())

```

Comment:
If the only difference between the four copies of the recipe-loading code is the base type of the recipes, they should be moved to New() on the parent type, with a var for the recipe type to load.